core/templates/domain/question/question-update.service.ts

```diff
@@ -128,19 +90,17 @@ export class QuestionUpdateService {
   }
 
   _updateContentIdsInAssets(newState: State, oldState: State): void {
-    let newContentIds = this._getAllContentIds(newState);
-    let oldContentIds = this._getAllContentIds(oldState);
+    let newContentIds = new Set(newState.getAllContentIds());
+    let oldContentIds = new Set(oldState.getAllContentIds());
```

Comment:
Why do we need to do a Set operation here? Are there scenarios where we would have duplicate content IDs?